geyser-plugin-interface/src/geyser_plugin_interface.rs

```diff
@@ -318,6 +319,12 @@ pub trait GeyserPlugin: Any + Send + Sync + std::fmt::Debug {
         Ok(())
     }
 
+    /// Called when a bank is updated
+    #[allow(unused_variables)]
+    fn update_bank(&self, bank: Option<Arc<Bank>>, status: SlotStatus) -> Result<()> {
```

Comment:
I think we need bank_notifications_enabled interface

Reply:
I am nervous about exposing the Bank to the plugin. How do you plan to use the Bank?

Reply:
Bank is declared in runtime and not even in the SDK and it will be subjected to heavy internal documentations. I need to understand the justification more.

Reply:
The main issue is that if we are truly serious about offering more of the functionalities of the RPC interface via Geyser so that alternative interfaces can be design (e.g. grpc) then a way to get accounts snapshots is necessary. The only other option is to track all state in the plugin which means that for many use cases you are tracking all state both in plugin and in validator. Having access to snapshot state is necessary for many applications.

To give one example: if we are going to service streams over gRPC we need a way for most applications t deliver an initial state at the start of the new stream and a snapshot state at regular intervals (most applications implement some kind of logic like this).

Since the plugin cannot ahead of time know which accounts may or may not be requested by the client the workarounds for having the ability to service initial state when a client connects would be:

a) Store all state for all accounts and keep them updated. Then service new connections from this state. This leads to a lot of resource waste and a much greater complexity for the plugin.

b) Require clients to combine using the traditional RPC interface to receive state snapshot and then use geyser to keep it updated. This leads to synchronisation complexity in the client plus makes clients more complex since they need to integrate with multiple RPC stacks. It also means that we cannot improve upon the JSON RPC interface in efficiency since we require clients to fall back upon the built in JSONRPC interface.

Sample use cases have been brought up by many app developers, including  Mangos lite rpc client that could be made to depend entirely on geyser/grpc.

If I am missing some option, let me know, but as far as I can tell we either say that we dont want the Geyser interface to be able to offer a replacement for standard RPC or we add a lot of complexity to the plugins and/or clients. 

If there is a better way to expose account state snapshots I am all ears.